lib/ansible/modules/cloud/memset/memset_zone_record.py

```diff
@@ -0,0 +1,363 @@
+    may cause some changes to occur as the defaults are enforced (if
+    the user has only configured required variables).
+    '''
+    has_changed, has_failed = False, False
+    msg, memset_api = None, None
+
+    # assemble the new record.
+    new_record = dict()
+    new_record['zone_id'] = zone_id
+    for arg in ['priority', 'address', 'relative', 'record', 'ttl', 'type']:
+        new_record[arg] = args[arg]
+
+    # if we have any matches, update them.
+    if records:
+        for zone_record in records:
+            # record exists, add ID to payload.
+            new_record['id'] = zone_record['id']
+            if zone_record == new_record:
+                # nothing to do; record is already correct so we populate
+                # the return var with the existing record's details.
+                memset_api = zone_record
+                return(has_changed, has_failed, memset_api, msg)
+            else:
+                # merge dicts ensuring we change any updated values
+                payload = zone_record.copy()
+                payload.update(new_record)
+                api_method = 'dns.zone_record_update'
+                if args['check_mode']:
+                    has_changed = True
+                    # return the new record to the user in the returned var.
+                    memset_api = new_record
+                    return(has_changed, has_failed, memset_api, msg)
+                has_failed, msg, response = memset_api_call(api_key=args['api_key'], api_method=api_method, payload=payload)
+                if not has_failed:
+                    has_changed = True
+                    memset_api = new_record
+                    # empty msg as we don't want to return a boatload of json to the user.
+                    msg = None
+    else:
+        # no record found, so we need to create it
+        api_method = 'dns.zone_record_create'
+        payload = new_record
+        if args['check_mode']:
+            has_changed = True
+            # populate the return var with the new record's details.
+            memset_api = new_record
+            return(has_changed, has_failed, memset_api, msg)
+        has_failed, msg, response = memset_api_call(api_key=args['api_key'], api_method=api_method, payload=payload)
+        if not has_failed:
+            has_changed = True
+            memset_api = new_record
+            #  empty msg as we don't want to return a boatload of json to the user.
+            msg = None
+
+    return(has_changed, has_failed, memset_api, msg)
+
+
+def delete_zone_record(args=None, records=None, payload=None):
+    '''
+    Matching records can be cleanly deleted without affecting other
+    resource types, so this is pretty simple to achieve.
+    '''
+    has_changed, has_failed = False, False
+    msg, memset_api = None, None
+
+    # if we have any matches, delete them.
+    if records:
+        for zone_record in records:
+            if args['check_mode']:
```

Comment:
not sure if this works, did you test check mode? shouldn't this be `module.check_mode`?

Reply:
I add it to the args [here](https://github.com/ansible/ansible/pull/40392/files/c6299ea97e6f28f4bc9fc5c6089f4fff088e2965#diff-f980ee4d7a8fcc19fa363523d47d03c7R349)

Reply:
ok, missed that!